src/core/meshcop/dataset_manager.cpp

```diff
@@ -380,6 +395,16 @@ ThreadError DatasetManager::Set(Coap::Header &aHeader, Message &aMessage, const
              tlvType == Tlv::kPanId || tlvType == Tlv::kNetworkMasterKey))
         {
             isUpdateAffectConnectivity = true;
+
+            if (channel.GetChannel() != mNetif.GetMac().GetChannel() ||
+                panId.GetPanId() != mNetif.GetMac().GetPanId() ||
+                memcmp(meshLocalPrefix.GetMeshLocalPrefix(), mNetif.GetMle().GetMeshLocalPrefix(),
+                       meshLocalPrefix.GetLength()) != 0 ||
+                memcmp(masterKey.GetNetworkMasterKey(), mNetif.GetKeyManager().GetMasterKey(NULL),
+                       masterKey.GetLength()) != 0)
+            {
+                isUpdateIdenticalValue = false;
```

Comment:
Considering that MGMT_ACTIVE_SET.req should not include tlv that would affect connectivity from commissioner, so here we can use `isUpdateAffectConnectivity` and `isUpdateIdenticalValue` to indicate a true MGMT_ACTIVE_SET.req including tlv that will affect connectivity. This check is also updated in line 447.